diffsims/generators/rotation_list_generators.py

```diff
@@ -159,83 +161,521 @@ def get_grid_around_beam_direction(beam_rotation, resolution, angular_range=(0,
+    """
+    # normalize the vertex vectors
+    vertices = normalize_vectors(vertices)
+    nn1_vec = _get_first_nearest_neighbors(vertices, leaf_size)
+    # the dot product between each point and its nearest neighbor
+    nn_dot = np.sum(vertices*nn1_vec, axis=1)
+    # angles
+    angles = np.rad2deg(np.arccos(nn_dot))
+    return angles
+
+
+def _get_max_grid_angle(vertices, leaf_size=50):
+    """
+    Helper function to get the maximum angle between nearest neighbor grid
+    points on a grid.
+    """
+    return np.max(_get_angles_between_nn_gridpoints(vertices, leaf_size))
+
+
+def _get_min_grid_angle(vertices, leaf_size=50):
+    """
+    Helper function to get the minimum angle between nearest neighbor grid
+    points on a grid.
+    """
+    return np.min(_get_angles_between_nn_gridpoints(vertices, leaf_size))
+
+
+def get_icosahedral_mesh_vertices(resolution):
+    """
+    Return the (x, y, z) coordinates of the vertices of an icosahedral
+    mesh of a cube, see reference [1]. Method was adapted from meshzoo [2].
+
+    Parameters
+    ----------
+    resolution : float
+        The maximum angle in degrees between neighboring grid points. Since
+        the mesh is generated iteratively, the actual maximum angle in the
+        mesh can be slightly smaller.
+
+    Returns
+    -------
+    points_in_cartesian : np.array (N,3)
+        Rows are x, y, z where z is the 001 pole direction
+
+    References
+    ----------
+    [1] https://medium.com/game-dev-daily/four-ways-to-create-a-mesh-for-a-sphere-d7956b825db4
+    [2] https://github.com/nschloe/meshzoo/blob/master/meshzoo/sphere.py
+    """
+    t = (1.0 + np.sqrt(5.0)) / 2.0
+    corners = np.array(
+        [
+            [-1, +t, +0],
+            [+1, +t, +0],
+            [-1, -t, +0],
+            [+1, -t, +0],
+            #
+            [+0, -1, +t],
+            [+0, +1, +t],
+            [+0, -1, -t],
+            [+0, +1, -t],
+            #
+            [+t, +0, -1],
+            [+t, +0, +1],
+            [-t, +0, -1],
+            [-t, +0, +1],
+        ]
+    )
+    faces = [
+        (0, 11, 5),
+        (0, 5, 1),
+        (0, 1, 7),
+        (0, 7, 10),
+        (0, 10, 11),
+        (1, 5, 9),
+        (5, 11, 4),
+        (11, 10, 2),
+        (10, 7, 6),
+        (7, 1, 8),
+        (3, 9, 4),
+        (3, 4, 2),
+        (3, 2, 6),
+        (3, 6, 8),
+        (3, 8, 9),
+        (4, 9, 5),
+        (2, 4, 11),
+        (6, 2, 10),
+        (8, 6, 7),
+        (9, 8, 1),
+    ]
+    n = 1
+    angle = _get_max_grid_angle(corners)
+    # maybe not the most efficient approach, but the least work
+    while(angle > resolution):
+        vertices = _compose_from_faces(corners, faces, n)
+        angle = _get_max_grid_angle(vertices)
+        n = n+1
+    # push all nodes to the sphere
+    norms = np.sqrt(np.einsum("ij,ij->i", vertices, vertices))
+    vertices = (vertices.T / norms.T).T
+    return vertices
+
+
+def _vector_grid_to_euler(vectors):
```

Comment:
I think this is unused?

Reply:
Not in the other functions, but I used it for converting my grid to euler angles which I export so I could import it to MTEX. I couldn't find a suitable alternative elsewhere, maybe there is already a function doing this? If not, I'm not sure where it fits best, maybe not in this file. If you think it should stay, I would say it's more of a public function.

Reply:
For now I have left it in here renamed, made it a public function, added a better docstring, and added a unit test for it. If at some point things get reshuffled and there are better places to put this in feel free to move/delete it.